src/main/java/io/anserini/qa/passage/WmdPassageScorer.java

```diff
@@ -0,0 +1,142 @@
+import org.json.JSONObject;
+
+import java.io.*;
+import java.util.*;
+
+public class WmdPassageScorer implements PassageScorer {
+
+  private final WordEmbeddingDictionary wmdDictionary;
+  private final MinMaxPriorityQueue<ScoredPassage> scoredPassageHeap;
+  private final int topPassages;
+  private final List<String> stopWords;
+
+  public WmdPassageScorer(String index, Integer k) throws IOException {
+    this.wmdDictionary = new WordEmbeddingDictionary(index);
+    wmdDictionary.getEmbeddingVector("newton");
+    this.topPassages = k;
+    scoredPassageHeap = MinMaxPriorityQueue.maximumSize(topPassages).create();
+    stopWords = new ArrayList<>();
+
+    //Get file from resources folder
+    InputStream is = getClass().getResourceAsStream("/io/anserini/qa/english-stoplist.txt");
+    BufferedReader bRdr = new BufferedReader(new InputStreamReader(is));
+    String line;
+    while ((line = bRdr.readLine()) != null) {
+      if (!line.contains("#")) {
+        stopWords.add(line);
+      }
+    }
+  }
+
+  public double distance(float[] leftVector, float[] rightVector) {
+    double sum = 0.0;
+
+    for (int i = 0; i < leftVector.length; i++) {
+      sum += Math.pow(leftVector[i] - rightVector[i], 2);
+    }
+    return  Math.sqrt(sum);
+  }
+
+  @Override
+  public void score(String query, Map<String, Float> sentences) throws Exception {
+    StandardAnalyzer sa = new StandardAnalyzer();
+    TokenStream tokenStream = sa.tokenStream("contents", new StringReader(query));
+    CharTermAttribute charTermAttribute = tokenStream.addAttribute(CharTermAttribute.class);
+    tokenStream.reset();
+
+    HashSet<String> questionTerms = new HashSet<>();
+    HashSet<String> candidateTerms = new HashSet<>();
+
+    // avoid duplicate passages
+    HashSet<String> seenSentences = new HashSet<>();
+
+    while (tokenStream.incrementToken()) {
+      questionTerms.add(charTermAttribute.toString());
+    }
+
+    for (Map.Entry<String, Float> sent : sentences.entrySet()) {
+      double wmd = 0.0;
+      candidateTerms.clear();
+      StandardAnalyzer sa2 = new StandardAnalyzer();
+      TokenStream candTokenStream = sa2.tokenStream("contents", new StringReader(sent.getKey()));
+      charTermAttribute = candTokenStream.addAttribute(CharTermAttribute.class);
+      candTokenStream.reset();
+
+      while (candTokenStream.incrementToken()) {
+        candidateTerms.add(charTermAttribute.toString());
+      }
+
+      for(String qTerm : questionTerms) {
+        double minWMD = Double.MAX_VALUE;
+        for (String candTerm : candidateTerms) {
+          try {
+            double thisWMD = distance(wmdDictionary.getEmbeddingVector(qTerm), wmdDictionary.getEmbeddingVector(candTerm));
+            if (minWMD > thisWMD) {
+              minWMD = thisWMD;
+            }
+          } catch (ArrayIndexOutOfBoundsException e) {
+            // term is OOV
+          }
+        }
+        if (minWMD != Double.MAX_VALUE)
```

Comment:
add brackets to the `if` statement, even if it is only one line... be consistent